### PR TITLE
[Scoop] Added scoop-license badge.

### PR DESCRIPTION
**Pull request by MohanKumarAmbati**
Hi Team,

Added scoop-license badge.

closes - #10297

### CODE REVIEW

**Comment by chris48s** [comment]
There is a lot of code here which is straight copy/pasted from the scoop version service.

I think if we're going to merge this I would like to see some of the common code extracted into a shared helper with a `ScoopBase` class which both `ScoopLicense` and `ScoopVersion` inherit from.

Specifically, several of the schemas and bucket file cache can be shared and nearly all of the logic in `handle()` can be extracted to a shared `async fetch()` method on `ScoopBase`.

**Comment by chris48s** [comment]
Running the existing service tests for scoop version should help you ensure we're not breaking the existing code as you refactor. You should also be able to look at some other services where we have a base class and multiple services extending it to see some examples. If you get stuck, feel free to ask for help.

**Comment by MohanKumarAmbati** [comment]
Chris, refactored into scoop-base according to the suggestions.

File: services/scoop/scoop-license.service.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    pattern: ':app',
+    queryParamSchema,
+  }
+
+  static openApi = {
+    '/scoop/l/{app}': {
+      get: {
+        summary: 'Scoop License',
+        description:
+          '[Scoop](https://scoop.sh/) is a command-line installer for Windows',
```
[the remark this answers is not in the document]
Comment (MohanKumarAmbati) [reply services/scoop/scoop-license.service.js]: Yeah done with the changes @jNullj, can you please review the changes.

**Review by jNullj — CHANGES_REQUESTED** [review]
It might be a bit more elegant to use `createServiceTester` here, as we only export one service in the class.

**Comment by MohanKumarAmbati** [comment]
> removed

Didn't get you, Jonathan.

**Comment by jNullj** [comment]
> > removed
> 
> Didn't get you, Jonathan.

I removed that comment, it was made by mistake and you can ignore it. Sorry for not writing that more clearly.
Wrote that as there was no option to delete it.
Please see the suggestions added after my review.

**Comment by MohanKumarAmbati** [comment]
> > > removed
> > 
> > Didn't get you, Jonathan.
> 
> I removed that comment, it was made by mistake and you can ignore it. Sorry for not writing that more clearly.
> Wrote that as there was no option to delete it.
> Please see the suggestions added after my review. 

Made changes - refactoring of description. Am I missing something else, please let me know if any.

File: services/scoop/scoop-base.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { NotFound } from '../index.js'
+
+const gitHubRepoRegExp =
+  /https:\/\/github.com\/(?<user>.*?)\/(?<repo>.*?)(\/|$)/
+
+const bucketsSchema = Joi.object()
+  .pattern(/.+/, Joi.string().pattern(gitHubRepoRegExp).required())
+  .required()
+
+export class ScoopBase extends ConditionalGithubAuthV3Service {
```
[the remark this answers is not in the document]
Comment (jNullj) [reply services/scoop/scoop-base.js:13]: i think this should move to base and was left out in the refactor

**Review by jNullj — CHANGES_REQUESTED** [review]
Didn't notice the review was in draft, now you can see my review suggestions.

**Review by jNullj — CHANGES_REQUESTED** [review]
Also this one

**Comment by MohanKumarAmbati** [comment]
Got it, will work on that.

File: services/scoop/scoop-license.tester.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+t.create('License (valid) - with nested response')
+  .get('/l/ngrok.json')
```
Comment (jNullj) [inline services/scoop/scoop-license.tester.js:5]: you don't need the base route when using `createServiceTester`
fox example here `/l/ngrok.json` can be only `/ngrok.json`

Comment (MohanKumarAmbati) [reply services/scoop/scoop-license.tester.js:5]: Just clicked on resolve and that triggered the build. Haven't made the changes. I'll let you know once I'm done, Johnathan.

File: services/scoop/scoop-version.service.js [hunk 4]
```diff
@@ -1,23 +1,16 @@
...
 }).required()
 const queryParamSchema = Joi.object({
   bucket: Joi.string(),
 })
 
-export default class ScoopVersion extends ConditionalGithubAuthV3Service {
+export default class ScoopVersion extends ScoopBase {
   // The buckets file (https://github.com/lukesampson/scoop/blob/master/buckets.json) changes very rarely.
   // Cache it for the lifetime of the current Node.js process.
   buckets = null
```
[the remark this answers is not in the document]
Comment (MohanKumarAmbati) [reply services/scoop/scoop-version.service.js]: made changes in as part of the commit - https://github.com/badges/shields/pull/10627/commits/e7e40bb0f1af9bdd19cefde1ed43016720ebfcc5

File: services/scoop/scoop-license.tester.js (hunk 3 above)
Comment (MohanKumarAmbati) [reply services/scoop/scoop-license.tester.js]: made changes as part of - https://github.com/badges/shields/pull/10627/commits/e7e40bb0f1af9bdd19cefde1ed43016720ebfcc5

**Comment by MohanKumarAmbati** [comment]
@jNullj , changes are done accordingly. Can you please review it once and let me know if there are any additional changes required.

**Review by jNullj — APPROVED** [review]
LGTM, thank you for taking the time and applying these changes.
image

**Comment by chris48s** [comment]
Thanks for picking up the review on this one @jNullj 

I've just had a look over this. I have one further request for an improvement: Can we move `queryParamSchema` to `scoop-base.js` and import it in both the version and license files please.

After, its good to go.

Nice work, both.

**Comment by MohanKumarAmbati** [comment]
> I've just had a look over this. I have one further request for an improvement: Can we move `queryParamSchema` to `scoop-base.js` and import it in both the version and license files please.

- Changes are made accordingly - https://github.com/badges/shields/pull/10627/commits/ed5bad09d2a9f5a25cee241ee16cb88da8bb736e.

**Review by chris48s — APPROVED** [review]
thanks

**Pull request merged**